### DISCUSSION TITLE
Is is possible to use pystack with MacOS?

### DISCUSSION THREAD

**Discussion by binary-signal** [bloomberg/pystack#104] [Q&A] [answered]
I was wondering if it's possible to use pystack in MacOS, not sure if some of the dependencies make sense in MacOS for example elfutils.

Is there any plans to provide support in MacOS like in memray?

Maybe if I rephrase the question is it technically possible to run pystack in MacOS ?

**Comment by pablogsal** [comment, marked as answer]
Hi @binary-signal and thanks for the question!

Unfortunately `pystack` doesn't work with MacOS as we are entering all our efforts in Linux platforms. This doesn't mean is not technically possible, but it would be a huge amount of work as we would need to code a lot of the backends for macOS, including mach-o analysis, memory copying and core file support. 

If you need similar functionality in MacOS you can check out [Austin](https://github.com/P403n1x87/austin) or [py-spy](https://github.com/benfred/py-spy). Both tools are profilers but they offer a `dump` functionality that's very similar to `pystack`.

> Is there any plans to provide support in MacOS like in memray?

Not at the moment, but it may be possible in the future as we did with `memray` :)
